rskj-core/src/test/java/co/rsk/peg/BridgeSupportTestUtil.java

```diff
@@ -22,6 +24,53 @@ public static Repository createRepository() {
         return new MutableRepository(new MutableTrieCache(new MutableTrieImpl(null, new Trie())));
     }
 
+    public static PartialMerkleTree createValidPmtWithTransactions(List<Sha256Hash> hashesToAdd, NetworkParameters networkParameters) {
+        byte[] bits = new byte[1];
+        bits[0] = 0x3f;
+
+        return new PartialMerkleTree(networkParameters, bits, hashesToAdd, 1);
+    }
+
+    public static void recreateChainAtHeightWithStoredBlockWithPmtAtHeight(
+        BtcBlockStoreWithCache btcBlockStoreWithCache,
+        int chainHeight,
+        PartialMerkleTree partialMerkleTree,
+        int btcBlockWithPmtHeight,
+        NetworkParameters networkParameters
+    ) throws BlockStoreException {
+
+        // first create a block that has the wanted partial merkle tree
+        BtcBlock btcBlockWithPmt = createBtcBlockWithPmt(partialMerkleTree, networkParameters);
+        // store it on the chain at wanted height
+        StoredBlock storedBtcBlockThatHasTransactions = new StoredBlock(btcBlockWithPmt, BigInteger.ONE, btcBlockWithPmtHeight);
+        btcBlockStoreWithCache.put(storedBtcBlockThatHasTransactions);
+        btcBlockStoreWithCache.setMainChainBlock(btcBlockWithPmtHeight, btcBlockWithPmt.getHash());
+
+        // create and store a new chainHead at wanted chain height
+        Sha256Hash randomTransactionHash = Sha256Hash.of(Hex.decode("aa"));
```

Comment:
This is not really random. I would suggest to either rename it or make it really random.

Reply:
i named it _random_ just because it is not related with the values received as inputs

Reply:
Maybe using `otherTxHash` would clarify things